pyproject.toml

```diff
@@ -29,7 +30,6 @@ classifiers = [
 
 [project.optional-dependencies]
 dev = [
-    "black>=22.0,<23.12",  # A deterministic code formatter
```

Comment:
There's some `tool.black` config below that should be removed, and some `tool.ruff.format` config that should be brought over from PUDL.